add latest as default tag for container images

Ferada opened the pull request:
Fix for <https://github.com/anchore/syft/issues/411>, according to [this comment](https://github.com/anchore/syft/issues/411#issuecomment-897676650), not sure that's the right spot though.

Also, this method will then resolve the container name to the fully expanded version, including the registry, I'm not sure that doesn't interfere with anything, just to mention it.

The OCI version already works, no change needed there.

spiffcs commented:
Thanks for the PR @Ferada!

wagoodman reviewed (approved):
Thanks! I added a few tests and changed the behavior some to not assume a registry -- we try not to make any assumptions based on the input and allow the tooling (in this case docker) to resolve any defaults here (notable exception is the tag of course).

Ferada commented:
> I added a few tests and changed the behavior some to not assume a registry -- we try not to make any assumptions based on the input and allow the tooling (in this case docker) to resolve any defaults here (notable exception is the tag of course).

Makes sense, thanks.
